tests/unit/amalgamation-mixin.spec.ts

```diff
@@ -0,0 +1,251 @@
+    expect(wrapper.vm.foreignUnlimited({ type: AmlTypes.FOREIGN })).toBeNull()
+    vi.spyOn(wrapper.vm, 'isAnyBcCompany', 'get').mockReturnValue(true)
+
+    // verify not ULC only
+    store.setEntityType(null)
+    expect(wrapper.vm.foreignUnlimited({ type: AmlTypes.FOREIGN })).toBeNull()
+    store.setEntityType(CorpTypeCd.BC_ULC_COMPANY)
+  })
+
+  it('correctly evaluates "cccMismatch" rule', () => {
+    // init
+    store.setEntityType(null)
+
+    // verify rule
+    expect(wrapper.vm.cccMismatch({ type: AmlTypes.LEAR, legalType: CorpTypeCd.BC_CCC })).toBe(AmlStatuses.ERROR_CCC_MISMATCH)
+
+    // verify not LEAR only
+    expect(wrapper.vm.cccMismatch({ type: AmlTypes.FOREIGN, legalType: CorpTypeCd.BC_CCC })).toBeNull()
+
+    // verify not legalType only
+    expect(wrapper.vm.cccMismatch({ type: AmlTypes.LEAR, legalType: null })).toBeNull()
+
+    // verify not CCC only
+    store.setEntityType(CorpTypeCd.BC_CCC)
+    expect(wrapper.vm.cccMismatch({ type: AmlTypes.LEAR, legalType: CorpTypeCd.BC_CCC })).toBeNull()
+  })
+
+  it('correctly evaluates "foreignUnlimited2" rule', () => {
+    // init
+    vi.spyOn(wrapper.vm, 'isAnyForeign', 'get').mockReturnValue(true)
+    store.setEntityType(CorpTypeCd.BC_ULC_COMPANY)
+
+    // verify rule
+    expect(wrapper.vm.foreignUnlimited2({ type: AmlTypes.LEAR, legalType: CorpTypeCd.BC_COMPANY })).toBe(AmlStatuses.ERROR_FOREIGN_UNLIMITED2)
+
+    // verify not LEAR only
+    expect(wrapper.vm.foreignUnlimited2({ type: AmlTypes.FOREIGN, legalType: CorpTypeCd.BC_COMPANY })).toBeNull()
+
+    // verify not legalType only
+    expect(wrapper.vm.foreignUnlimited2({ type: AmlTypes.LEAR, legalType: null })).toBeNull()
+
+    // verify not ULC only
+    store.setEntityType(null)
+    expect(wrapper.vm.foreignUnlimited2({ type: AmlTypes.LEAR, legalType: CorpTypeCd.BC_COMPANY })).toBeNull()
+  })
+
+  it('correctly evaluates "xproUlcCcc" rule - for ULC', () => {
+    // init
+    store.setEntityType(CorpTypeCd.BC_ULC_COMPANY)
+
+    // verify rule
+    expect(wrapper.vm.xproUlcCcc({ type: AmlTypes.FOREIGN })).toBe(AmlStatuses.ERROR_XPRO_ULC_CCC)
+
+    // verify not FOREIGN only
+    expect(wrapper.vm.xproUlcCcc({ type: AmlTypes.LEAR })).toBeNull()
+
+    // verify not ULC or CCC only
+    store.setEntityType(null)
+    expect(wrapper.vm.xproUlcCcc({ type: AmlTypes.FOREIGN })).toBeNull()
+  })
+
+  it('correctly evaluates "xproUlcCcc" rule - for CCC', () => {
+    // init
+    store.setEntityType(CorpTypeCd.BC_CCC)
+
+    // verify rule
+    expect(wrapper.vm.xproUlcCcc({ type: AmlTypes.FOREIGN })).toBe(AmlStatuses.ERROR_XPRO_ULC_CCC)
+
+    // verify not FOREIGN only
+    expect(wrapper.vm.xproUlcCcc({ type: AmlTypes.LEAR })).toBeNull()
+
+    // verify not ULC or CCC only
+    store.setEntityType(null)
+    expect(wrapper.vm.xproUlcCcc({ type: AmlTypes.FOREIGN })).toBeNull()
+  })
+
+  it('correctly evaluates "foreignUnlimited3" rule', () => {
```

Comment:
I just noticed that we're not using this rule (we don't add it to the array). Going to double check again if this is needed.

Reply:
Whoops! I will add it.

Reply:
Fixed, thanks! Resolving.

Reply:
Done.

You may have noticed that some rules are from the point of view of a foreign company, while this rule (for example) is from the POV of a ULC. This results in messages on both TINGs, but they go away when the issue is resolved.

Another thing is that there was (and still is) some debate on the rules requirements doc and whether there are superset rules or duplicate rules. In the end I stopped worrying about it and just coded what the doc said.

Reply:
Yes, makes sense.

Deal, thanks Sev!

Reply:
Always listen to your doc 🍎 
😆